### DISCUSSION TITLE
Save a Remote to a Repository's config

### DISCUSSION THREAD

**Discussion by skeggse** [GitoxideLabs/gitoxide#1561] [Q&A]
Is there an easy way to save a `Remote` to a `Repository`? It looks like `gix` does it under the hood with `write_remote_to_local_config_file` during fetch, but that's in a private module.

It would seem like `remote.save_as_to("origin".into(), repo.config_snapshot_mut().config)` should work, except `config` is crate-private. Is there an accepted way to do this?

**Comment by Byron** [comment]
Indeed, the `save_as_to` or `save_to` methods can be used to write remote configuration, and it is possible to obtain a bare `gix_config::File` by means of `&mut snapshot` I think.

However, writing the configuration back isn't easy, non-plumbing APIs don't exist for that yet.
